queries.sql

```diff
@@ -42,4 +42,26 @@ SELECT MIN(species) FROM animals GROUP BY species;
 SELECT MAX(species) FROM animals GROUP BY species;
 SELECT AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN Jan 1, 1990 AND Dec 31, 2000 GROUP BY species;
 
+-- Write queries (using JOIN) :
+-- What animals belong to Melody Pond?
+SELECT animals.name FROM animals JOIN owners ON animals.owner_id = owners.id WHERE full_name='Melody Pond';
 
+--List of all animals that are pokemon (their type is Pokemon).
+SELECT animals.name FROM animals JOIN species ON animals.species_id=species.id WHERE species_id=1;
+
+--List all owners and their animals, remember to include those that don't own any animal.
+SELECT animals.name,full_name FROM owners LEFT JOIN animals ON animals.owner_id=owners.id;
+
+--How many animals are there per species?
+SELECT species.name, COUNT(*) from animals JOIN species ON animals.species_id=species.id GROUP BY species.name;
+
+--List all Digimon owned by Jennifer Orwell.
+SELECT animals.name FROM animals JOIN owners ON animals.owner_id=owners.id WHERE owners.full_name='Jennifer Orwell' AND species_id=2;
+
+--List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT animals.name FROM animals JOIN owners ON animals.owner_id=owners.id 
+WHERE owners.full_name='Dean Winchester' AND escape_attempts=0;
```

Comment:
- [ ]  The query here is to list all animals owned by Dean Winchester that haven't tried to escape which is meant to return all details about those animals but it is currently returning just the names. e.g.
```sql
 SELECT animals.* FROM animals
   LEFT JOIN owners
   ON animals.owner_id = owners.id
   WHERE owners.full_name = 'Dean Winchester'
   AND animals.escape_attempts = 0;
```